rpm.go

```diff
@@ -496,39 +491,108 @@ func (r *RPM) writeFileIndexes(h *index) {
 	h.Add(tagFileLangs, EntryStringSlice(fileLangs))
 }
 
-// AddPretrans adds a pretrans scriptlet
+func (r *RPM) writeScriptlets(h *index) {
+	for name, s := range r.scriptlets {
+		if s.content != "" || s.interpreter != "" {
+			tagInfo, found := scriptletsTypes[name]
+			if !found {
+				panic(fmt.Sprintf("internal error: invalid scriptlet name %q", name))
+			}
+			// If only an interpreter is given then the `scriptlet` becomes a `program` which is called without arguments
+			if s.content != "" {
+				h.Add(tagInfo.tag, EntryString(s.content))
+			}
+			if s.interpreter == "" {
```

Comment:
I think the "positive" logic would easier to follow here. Also avoid the extra indentation by using an else if.

"if the interpreter is not empty, use it. Else if it's lua, then use lua. Else use the default"